### ISSUE TITLE
Batch processing gui for "Export layers to DXF" is missing functionality for loading layers into separate rows in the table

### ISSUE THREAD

**Issue by AlisterH** [qgis/QGIS#47325]
### What is the bug or the crash?

BACKGROUND COMMENTS
The "Export layers to DXF" algorithm is relatively unusual (but not unique) because it has a single "input layers" input where more than one input layer can be selected.  People are probably relatively unlikely to need to "run as batch process".  
ISSUE
If people do need to "run as batch process", for some reason this algorithm is missing this functionality that is provided with other algorithms:
```
Add Files by Pattern...
Select Files...
Add All Files from a Directory...
Select from Open Layers...
````
Maybe someone thought this functionality isn't necessary for this algorithm?  
![image](https://user-images.githubusercontent.com/714600/153734363-f663b01b-db11-4c1d-b1c3-fe55bf9e1186.png)
Compare with the vector translate algorithm:
![image](https://user-images.githubusercontent.com/714600/153734477-7d01d59f-f1b4-49cd-be70-818f9798201a.png)
And with r.patch, which also has a single "input layers" input where more than one input layer can be selected:
![image](https://user-images.githubusercontent.com/714600/153734449-71d437e9-54ee-489e-81f3-e430fc6b1db9.png)

### Steps to reproduce the issue

- click to open "Export layers to DXF" from the processing toolbox
- click "run as batch process" in the bottom left
- click "Autofill" button in the first column
The options are missing...

### Versions

QGIS version
3.22.3-Białowieża
QGIS code revision
1628765ec7
Qt version
5.15.2
Python version
3.9.5
GDAL/OGR version
3.4.1
PROJ version
8.2.1
EPSG Registry database version
v10.041 (2021-12-03)
GEOS version
3.10.0-CAPI-1.16.0
SQLite version
3.35.2
PDAL version
2.3.0
PostgreSQL client version
13.0
SpatiaLite version
5.0.1
QWT version
6.1.3
QScintilla2 version
2.11.5
OS version
Windows 10 Version 2009

Active Python plugins
AnotherDXF2Shape
1.2.3
AutoLayoutTool
1.1
autoSaver
2.6
batchvectorlayersaver
0.9
BulkVectorExport
1.1
CalculateGeometry
0.6.4
changeDataSource
3.1
DataPlotly
3.8.1
deactivate_active_labels
0.5
flowTrace
1.1.1
Generalizer3
1.0
GeoCoding
2.18
geo_sim_processing
1.2.0
getthemfiltered
0.1.3
gridSplitter
0.4.0
GroupStats
2.2.5
HideDocks
0.6.1
ImageServerConnector
2.1.1
ImportPhotos
3.0.3
joinmultiplelines
Version 0.4.1
karika
1.5
LayerBoard
1.0.1
linz-data-importer
2.2.3
loadthemall
3.3.0
MagicWand-master
1.3.1
mask
1.10.1
MemoryLayerSaver
4.0.4
mmqgis
2021.9.10
nominatim_locator_filter
0.2.4
numerator
0.2
numericalDigitize
0.4.6
pathfinder
version 0.4.1
plaingeometryeditor
3.0.0
plugin_reloader
0.9.1
powerpan
2.0
processing_saga
0.5.0
processing_taudem
3.0.0
processing_whitebox
0.14.0
profiletool
4.2.1
qchainage
3.0.1
QCopycanvas
0.5
qgis-plugin-findreplace-main
1
Qgis2threejs
2.6
QGIS3-getWKT
1.4
QuickMultiAttributeEdit3
version 3.0.3
QuickPrint
3.6.1
quicksaveqml
0.1.5
quick_map_services
0.19.27
rasmover-master
version 0.2
redLayer
2.2
selectThemes
3.0.1
simple_tools
0.4.1
splitmultipart
1.0.0
SplitPolygonShowingAreas
0.13
statist
3.2
themeselector
3.2.2
valuetool
3.0.15
ViewshedAnalysis
1.7
volume_calculation_tool
0.4
WaterNetAnalyzer-master
1.7
grassprovider
2.12.99
processing
2.12.99
sagaprovider
2.12.99

### Supported QGIS version

- [X] I'm running a supported QGIS version according to the roadmap.

### New profile

- [X] I tried with a new QGIS profile

**Comment by gioman** [comment]
> Maybe someone thought this functionality isn't necessary for this algorithm?

@AlisterH it is the other way: the tool in batch mode works in a different way compared to most of the others. Each line in btach mode allows you to export to a separate DXF, and each DXF can contain a number of layers that are loaded in the project. For each line you choose what layers (loaded in the project) you want to export in that particular output dxf

![image](https://user-images.githubusercontent.com/1951107/153746991-55eda6e8-5689-42e6-9608-d46034350fe1.png)

this is no bug to me.

**Comment by AlisterH** [comment]
Any chance we could get that "Feedback" label removed?

**Comment by gioman** [comment]
> Any chance we could get that "Feedback" label removed?

@AlisterH no, or not yet, because it is arguable that this a bug and the opinion (final word really) of a Processing developer/maintainer is needed.

**Comment by AlisterH** [comment]
And this is why I asked for that.  The assumption is that "Feedback" means the person who filed the ticket needs to provide more information...

**Comment by AlisterH** [comment]
Guess we'll call this a bump.

**Comment by AlisterH** [comment]
.

**Comment by AlisterH** [comment]
> If this issue should be reconsidered, please follow the guidelines in the previous comment _and reopen this issue_.

Hmmm.  Not sure how to reopen, so I guess that messaging needs to be improved.
And in this case nobody is "waiting on the submitter for answers", so it is a misuse of the feedback label anyway.

**Comment by alexbruy** [comment]
IMHO this should be a feature request, not bug. Export to DXF as well as some other algorothms don't use the same parameter type as "r.patch" and other algorithms which are mentioned as "working" here. That's why parameters behave differently.
